hibernate-core/src/test/java/org/hibernate/orm/test/query/NativeQueryLimitOffsetTest.java

```diff
@@ -91,6 +92,24 @@ public void testPartialLimitOffsetOnNativeQuery(SessionFactoryScope scope) {
 		);
 	}
 
+	@Test
+	@Jira( "https://hibernate.atlassian.net/browse/HHH-18309" )
+	public void testLimitOffsetZeroValue(SessionFactoryScope scope) {
+		scope.inTransaction( session -> {
+			List<Long> l = session.createNativeQuery( "select id from Person where name like :name", Long.class )
+					.setParameter( "name", "J%" )
+					.setFirstResult( 0 )
+					.getResultList();
+			assertEquals( 5, l.size() );
+
+			l = session.createNativeQuery( "select id from Person where name like :name", Long.class )
+					.setParameter( "name", "J%" )
+					.setMaxResults( 0 )
```

Comment:
zero `maxResults` will not trigger sql modification and execution, non-zero `maxResults` should be tested also.